typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -74,10 +77,17 @@ export abstract class HyperlaneDeployer<
     protected readonly factories: Factories,
     protected readonly options: DeployerOptions = {},
     protected readonly recoverVerificationInputs = false,
+    protected readonly icaAddresses = {},
   ) {
     this.logger = options?.logger ?? rootLogger.child({ module: 'deployer' });
     this.chainTimeoutMs = options?.chainTimeoutMs ?? 5 * 60 * 1000; // 5 minute timeout per chain
     this.options.ismFactory?.setDeployer(this);
+    if (Object.keys(icaAddresses).length > 0) {
```

Comment:
Remind me to make `objLength` and `isObjEmpty` util helpers one of these days. I'll get around to it :)